api/yarpcerrors/errors.go

```diff
@@ -79,87 +79,112 @@ func ErrorMessage(err error) string {
 func ResourceExhaustedErrorf(format string, args ...interface{}) error {
-	return newYARPCError(CodeResourceExhausted, "", format, args...)
+	return FromHeaders(CodeResourceExhausted, "", fmt.Sprintf(format, args...))
 }
 
 // FailedPreconditionErrorf returns a new yarpc error with code CodeFailedPrecondition.
 func FailedPreconditionErrorf(format string, args ...interface{}) error {
-	return newYARPCError(CodeFailedPrecondition, "", format, args...)
+	return FromHeaders(CodeFailedPrecondition, "", fmt.Sprintf(format, args...))
 }
 
 // AbortedErrorf returns a new yarpc error with code CodeAborted.
 func AbortedErrorf(format string, args ...interface{}) error {
-	return newYARPCError(CodeAborted, "", format, args...)
+	return FromHeaders(CodeAborted, "", fmt.Sprintf(format, args...))
 }
 
 // OutOfRangeErrorf returns a new yarpc error with code CodeOutOfRange.
 func OutOfRangeErrorf(format string, args ...interface{}) error {
-	return newYARPCError(CodeOutOfRange, "", format, args...)
+	return FromHeaders(CodeOutOfRange, "", fmt.Sprintf(format, args...))
 }
 
 // UnimplementedErrorf returns a new yarpc error with code CodeUnimplemented.
 func UnimplementedErrorf(format string, args ...interface{}) error {
-	return newYARPCError(CodeUnimplemented, "", format, args...)
+	return FromHeaders(CodeUnimplemented, "", fmt.Sprintf(format, args...))
 }
 
 // InternalErrorf returns a new yarpc error with code CodeInternal.
 func InternalErrorf(format string, args ...interface{}) error {
-	return newYARPCError(CodeInternal, "", format, args...)
+	return FromHeaders(CodeInternal, "", fmt.Sprintf(format, args...))
 }
 
 // UnavailableErrorf returns a new yarpc error with code CodeUnavailable.
 func UnavailableErrorf(format string, args ...interface{}) error {
-	return newYARPCError(CodeUnavailable, "", format, args...)
+	return FromHeaders(CodeUnavailable, "", fmt.Sprintf(format, args...))
 }
 
 // DataLossErrorf returns a new yarpc error with code CodeDataLoss.
 func DataLossErrorf(format string, args ...interface{}) error {
-	return newYARPCError(CodeDataLoss, "", format, args...)
+	return FromHeaders(CodeDataLoss, "", fmt.Sprintf(format, args...))
 }
 
 // UnauthenticatedErrorf returns a new yarpc error with code CodeUnauthenticated.
 func UnauthenticatedErrorf(format string, args ...interface{}) error {
-	return newYARPCError(CodeUnauthenticated, "", format, args...)
+	return FromHeaders(CodeUnauthenticated, "", fmt.Sprintf(format, args...))
+}
+
+// FromHeaders returns a new yarpc error from headers transmitted from the server side.
+//
+// If the specified code is CodeOK, this will return nil.
+// If the specified code is not CodeUnknown, this will not set the name field.
+//
+// This function should not be used by server implementations, use the individual
+// error constructors instead. This should only be uused by transport implementations.
+func FromHeaders(code Code, name string, message string) error {
```

Comment:
I want to explicitly discourage use of this - I feel like `FromHeaders` is self-documenting. If I did `New`, then users would use that function without reading the documentation.

The alternative is to not have this function at all, but then you end up recreating this functionality in every transport implementation.